GenerateCommunityList.cs

```diff
@@ -0,0 +1,125 @@
+
+            Auth auth = new Auth();
+            var graphAPIAuth = auth.graphAuth(log);
+
+            List<Community> communityList = await GetCommunityList(graphAPIAuth, log);
+
+            if (communityList != null)
+            {
+                var result = await StoreCommunityList(communityList, log);
+                return result;
+            }
+            else
+                return false;
+        }
+        public static async Task<List<Community>> GetCommunityList(GraphServiceClient graphServiceClient, ILogger log)
+        {
+            List<Community> communityList = new List<Community>();
+            Community community;
+            Owner owner;
+
+            try {
+                var groups = await graphServiceClient.Groups.Request(new List<QueryOption>() { new QueryOption("$count", "true") }).Header("ConsistencyLevel", "eventual").Filter("groupTypes/any(c:c eq 'Unified')").OrderBy("displayName asc").GetAsync();
+
+                foreach (var group in groups)
+                {
+                    community = new Community();
+                    community.DisplayName = group.DisplayName;
+
+                    var owners = await graphServiceClient.Groups[group.Id].Owners.Request().GetAsync();
+                    foreach (var o in owners)
+                    {
+                        var user = await graphServiceClient.Users[o.Id].Request().GetAsync();
+                        owner = new Owner();
+                        owner.DisplaName = user.DisplayName;
+                        owner.EmailAddress = user.Mail;
+                        community.OwnerInfo.Add(owner);
+                    }
+
+                    communityList.Add(community);
+                }
+            }
+            catch (Exception e) {
+                log.LogError($"Message: {e.Message}");
+                if (e.InnerException != null) log.LogError($"InnerException: {e.InnerException.Message}");
+                log.LogError($"StackTrace: {e.StackTrace}");
+                return null;
+            }
+
+            return communityList;
+        }
+
+        public static async Task<bool> StoreCommunityList(List<Community> communityList, ILogger log)
+        {
+            bool result = true;
+
+            IConfiguration config = new ConfigurationBuilder().AddJsonFile("appsettings.json", optional: true, reloadOnChange: true).AddEnvironmentVariables().Build();
+
+            string storageAccountName = config["storageAccountName"];
+            string containerName = config["containerName"];
+            string containerEndpoint = string.Format("https://{0}.blob.core.windows.net/{1}", storageAccountName, containerName);
+            string filename = $"community_list_{DateTime.UtcNow.Ticks}.json";
+
+            BlobContainerClient containerClient = new BlobContainerClient(new Uri(containerEndpoint), new DefaultAzureCredential());
+
+            byte[] byteArray = Encoding.ASCII.GetBytes(JsonConvert.SerializeObject(communityList));
```

Comment:
I'm not sure if this is the issue, but with have some issue with encoding. You can look at my tbs user, the é in my displayName is replace by ?